Add datasource for each Loki instance

alexandre-allard opened the pull request:
**Component**: salt,logs

**Context**: 

**Summary**:

**Acceptance criteria**:

alexandre-allard committed:
salt,logs: Deploy a Service per Loki Pod

These services allow to communicate which
each Pod individually, this is needed to
deploy datasources to target a specific
Loki instance.

alexandre-allard committed:
salt,logs: Deploy a Grafana datasource per Loki instance

This is a workaround to allow user to access a
specific Loki instance in case one would not
have all the logs (disk loss or node down for
a long period of time). This way the user can
select a healthy instance having all the data.

alexandre-allard committed:
salt,logs: Tell user that datasource can be changed

Display a message on Loki and Logs Grafana
dashboards to tell the user he can change
the datasource to target a specific Loki instance

bert-e commented:
# Hello alexandre-allard-scality,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/add-loki-datasources-for-each-instance` into
`development/2.11`
* [w/123.0/improvement/add-loki-datasources-for-each-instance](https://github.com/scality/metalk8s/compare/development/123.0...w/123.0/improvement/add-loki-datasources-for-each-instance) will be merged into `development/123.0` 



The following branches will **NOT** be impacted:

* `development/2.0`
* `development/2.1`
* `development/2.10`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`
* `development/2.7`
* `development/2.8`
* `development/2.9`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @chengyanjin

* @lucieleonard

* @jbwatenbergscality

* @alexandre-allard-scality

* @gdemonet

* @teddyandrieux

* @nicolast

* @thomasdanan

TeddyAndrieux commented on salt/metalk8s/addons/logging/loki/deployed/datasource.sls:
```diff
@@ -24,3 +33,11 @@ Deploy ConfigMap for Loki datasource:
               access: proxy
               url: http://loki.metalk8s-logging.svc:3100/
               version: 1
+{%- for index in range(loki.spec.deployment.replicas) %}
+            - name: Loki-{{ index }}
+              uid: metalk8s-loki-{{ index }}
+              type: loki
+              access: proxy
+              url: http://loki-{{ index }}.metalk8s-logging.svc:3100/
+              version: 1
+{%- endfor %}
```
Wondering if we can have one "default" datasource per type or if it's global, maybe we want to set the "Loki" datasource as default ?

gdemonet replied:
Not possible, we can only set one default per org

TeddyAndrieux replied:
:cry:

TeddyAndrieux reviewed (commented):
LGTM

Can you just add a changelog, and maybe we want to add a simple test just to be sure the "loki-x" service properly redirect to Loki
(I think about maybe one day a label in the Loki statefulset that change so the extra service we deploy does not get updated and we do not catch it)

alexandre-allard committed:
Add a CHANGELOG entry for Loki datasources

alexandre-allard committed:
tests: Add a test to ensure we can access Loki API

This test is to ensure we can access a specific Loki
instance through the Services.

bert-e commented:
# History mismatch



Merge commit #b3d3c0e71fee5a5958628ddb49e6231e2f73851b on the integration branch
`w/123.0/improvement/add-loki-datasources-for-each-instance` is merging a branch which is neither the current
branch `improvement/add-loki-datasources-for-each-instance` nor the development branch
`development/123.0`.

It is likely due to a rebase of the branch `improvement/add-loki-datasources-for-each-instance` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

TeddyAndrieux commented on CHANGELOG.md:
```diff
@@ -35,6 +35,9 @@
 
 ### Enhancements
 
+- Add each Loki instance as datasources in Grafana dashboards
+  (PR[#3681](https://github.com/scality/metalk8s/pull/3681))
+
```
You will need to rebase since this should be in the 2.11.1 section (yet another issue with changelog as a simple file)

alexandre-allard commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/add-loki-datasources-for-each-instance` into
`development/2.11`
* [w/123.0/improvement/add-loki-datasources-for-each-instance](https://github.com/scality/metalk8s/compare/development/123.0...w/123.0/improvement/add-loki-datasources-for-each-instance) will be merged into `development/123.0` 



The following branches will **NOT** be impacted:

* `development/2.0`
* `development/2.1`
* `development/2.10`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`
* `development/2.7`
* `development/2.8`
* `development/2.9`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @chengyanjin

* @lucieleonard

* @jbwatenbergscality

* @alexandre-allard-scality

* @gdemonet

* @teddyandrieux

* @nicolast

* @thomasdanan

bert-e commented:
# History mismatch



Merge commit #94c206485aa40be8a547e95d3aa65f02a385cbf7 on the integration branch
`w/123.0/improvement/add-loki-datasources-for-each-instance` is merging a branch which is neither the current
branch `improvement/add-loki-datasources-for-each-instance` nor the development branch
`development/123.0`.

It is likely due to a rebase of the branch `improvement/add-loki-datasources-for-each-instance` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

alexandre-allard commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/add-loki-datasources-for-each-instance` into
`development/2.11`
* [w/123.0/improvement/add-loki-datasources-for-each-instance](https://github.com/scality/metalk8s/compare/development/123.0...w/123.0/improvement/add-loki-datasources-for-each-instance) will be merged into `development/123.0` 



The following branches will **NOT** be impacted:

* `development/2.0`
* `development/2.1`
* `development/2.10`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`
* `development/2.7`
* `development/2.8`
* `development/2.9`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @chengyanjin

* @lucieleonard

* @jbwatenbergscality

* @alexandre-allard-scality

* @gdemonet

* @teddyandrieux

* @nicolast

* @thomasdanan

TeddyAndrieux reviewed (dismissed):
Ok for me !

bert-e commented:
# Conflict




There is a conflict between your branch `improvement/add-loki-datasources-for-each-instance` and the
destination branch `development/2.11`.

Please resolve the conflict on **the feature branch** (`improvement/add-loki-datasources-for-each-instance`).

```sh
git fetch && \
git checkout origin/improvement/add-loki-datasources-for-each-instance && \
git merge origin/development/2.11
```

Resolve merge conflicts and commit

```sh
git push origin HEAD:improvement/add-loki-datasources-for-each-instance
```

bert-e commented:
# History mismatch



Merge commit #bd94a75082ab8717356cff4c088fcccc3fb885e1 on the integration branch
`w/123.0/improvement/add-loki-datasources-for-each-instance` is merging a branch which is neither the current
branch `improvement/add-loki-datasources-for-each-instance` nor the development branch
`development/123.0`.

It is likely due to a rebase of the branch `improvement/add-loki-datasources-for-each-instance` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

alexandre-allard commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/add-loki-datasources-for-each-instance` into
`development/2.11`
* [w/123.0/improvement/add-loki-datasources-for-each-instance](https://github.com/scality/metalk8s/compare/development/123.0...w/123.0/improvement/add-loki-datasources-for-each-instance) will be merged into `development/123.0` 



The following branches will **NOT** be impacted:

* `development/2.0`
* `development/2.1`
* `development/2.10`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`
* `development/2.7`
* `development/2.8`
* `development/2.9`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @chengyanjin

* @lucieleonard

* @jbwatenbergscality

* @alexandre-allard-scality

* @gdemonet

* @teddyandrieux

* @nicolast

* @thomasdanan

alexandre-allard commented:
/approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.11`

* :heavy_check_mark: `development/123.0`



The following branches will **NOT** be impacted:

* `development/2.0`
* `development/2.1`
* `development/2.10`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`
* `development/2.7`
* `development/2.8`
* `development/2.9`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.11`

* :heavy_check_mark: `development/123.0`



The following branches have **NOT** changed:

* `development/2.0`
* `development/2.1`
* `development/2.10`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`
* `development/2.7`
* `development/2.8`
* `development/2.9`



Please check the status of the associated issue None.

Goodbye alexandre-allard-scality.
